src/test/rustdoc-ui/asm-foreign.rs

```diff
@@ -0,0 +1,19 @@
+// check-pass
+// Make sure rustdoc accepts asm! for a foreign architecture.
+
+#![feature(asm)]
+
+pub unsafe fn aarch64(a: f64, b: f64) {
+    let c;
+    asm!("add {:d}, {:d}, d0", out(vreg) c, in(vreg) a, in("d0") {
+        || {};
+        b
+    });
+    c
+}
```

Comment:
Can you test the output, instead of using `check-pass`? You'd move the test to `src/test/rustdoc` and add a `@has` annotation, see https://rustc-dev-guide.rust-lang.org/rustdoc-internals.html#dotting-is-and-crossing-ts. Just `@has asm_foreign/fn.aarch64.html` is probably fine.